src/all/mangafireto/src/eu.kanade.tachiyomi.extension.all.mangafireto/MangaFire.kt

```diff
@@ -0,0 +1,284 @@
+package eu.kanade.tachiyomi.extension.all.mangafireto
+
+import android.app.Application
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.select.Evaluator
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.net.URI
+
+open class MangaFire(
+    override val lang: String,
+) : ConfigurableSource, ParsedHttpSource() {
+    override val name = "MangaFire"
+
+    override val baseUrl = "https://mangafire.to"
+
+    override val supportsLatest = true
+
+    override val client = network.client.newBuilder()
+        .addInterceptor(MangaFireImageInterceptor)
+        .build()
+
+    private fun MangasPage.insertVolumeEntries(): MangasPage {
+        if (preferences.showVolume.not()) return this
+        val list = mangas.ifEmpty { return this }
+        val newList = ArrayList<SManga>(list.size * 2)
+        for (manga in list) {
+            val volume = SManga.create().apply {
+                url = manga.url + VOLUME_URL_SUFFIX
+                title = VOLUME_TITLE_PREFIX + manga.title
+                thumbnail_url = manga.thumbnail_url
+            }
+            newList.add(manga)
+            newList.add(volume)
+        }
+        return MangasPage(newList, hasNextPage)
+    }
```

Comment:
Use `flatMap`, it makes the code more easy to read.

```suggestion
    private fun MangasPage.insertVolumeEntries(): MangasPage {
        if (preferences.showVolume.not()) return this
        
        val newList = mangas.flatMap { manga ->
            val volume = SManga.create().apply {
                url = manga.url + VOLUME_URL_SUFFIX
                title = VOLUME_TITLE_PREFIX + manga.title
                thumbnail_url = manga.thumbnail_url
            }
          
            listOf(manga, volume)
        }
        
        return MangasPage(newList, hasNextPage)
    }
```